pkg/component/etcd/bootstrap/etcd_test.go

```diff
@@ -0,0 +1,76 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package bootstrap_test
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/component"
+	. "github.com/gardener/gardener/pkg/component/etcd/bootstrap"
+	secretsmanager "github.com/gardener/gardener/pkg/utils/secrets/manager"
+	fakesecretsmanager "github.com/gardener/gardener/pkg/utils/secrets/manager/fake"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+)
+
+var _ = Describe("Etcd", func() {
+	var (
+		c         client.Client
+		sm        secretsmanager.Interface
+		etcd      component.DeployWaiter
+		ctx       = context.Background()
+		namespace = "shoot--foo--bar"
+		image     = "some.registry.io/etcd:v3.5.10"
+		pod       = &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "etcd-main",
+				Namespace: namespace,
+			},
+		}
+	)
+
+	BeforeEach(func() {
+		c = fakeclient.NewClientBuilder().WithScheme(kubernetes.SeedScheme).Build()
+		sm = fakesecretsmanager.New(c, namespace)
+		etcd = New(c, namespace, sm, Values{Image: image})
+
+		By("Create secrets managed outside of this package for whose secretsmanager.Get() will be called")
+		Expect(c.Create(ctx, &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "ca-etcd", Namespace: namespace}})).To(Succeed())
+		Expect(c.Create(ctx, &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "ca-etcd-peer", Namespace: namespace}})).To(Succeed())
+	})
+
+	Describe("#Deploy", func() {
+		It("should successfully deploy bootstrap etcd", func() {
+			Expect(c.Get(ctx, client.ObjectKeyFromObject(pod), pod)).To(BeNotFoundError())
+			Expect(etcd.Deploy(ctx)).To(Succeed())
+			Expect(c.Get(ctx, client.ObjectKeyFromObject(pod), pod)).To(Succeed())
```

Comment:
Do you want to add some more assertions to ensure we create what we expect here?